packages/app/src/cli/services/function/build.ts

```diff
@@ -25,13 +25,22 @@ export async function buildJSFunction(fun: ExtensionInstance, options: JSFunctio
 }
 
 async function buildJSFunctionWithoutTasks(fun: ExtensionInstance, options: JSFunctionBuildOptions) {
-  options.stdout.write(`Building GraphQL types...\n`)
-  await buildGraphqlTypes(fun, options)
-  options.stdout.write(`Bundling JS function...\n`)
-  await bundleExtension(fun, options)
-  options.stdout.write(`Running javy...\n`)
-  await runJavy(fun, options)
-  options.stdout.write(`Done!\n`)
+  if (!options.signal?.aborted) {
```

Comment:
A little ugly but is there a better way?

Reply:
I'm a bit surprised by this, I would have expected the function to be killed when the abortcontroller was terminated. cc @matteodepalo

Reply:
I don't think there's anything magical about it, if you want to use `AbortController` in a custom way like this, you need to check the `aborted` flag or listen for the event --

https://blog.logrocket.com/complete-guide-abortcontroller-node-js/#implement-abortable-api-abortcontroller-abortsignal

Reply:
I think as @nickwesselman said, if you want to use AbortController here that should be the way to do it. The other way we've employed for UI extensions is to subscribe to the `abort` event and close the file watcher https://github.com/Shopify/cli/blob/main/packages/app/src/cli/services/dev/extension.ts#L114